src/power_grid_model/core/data_handling.py

```diff
@@ -151,6 +156,44 @@ def create_output_data(
             shape: Union[Tuple[int], Tuple[int, int]] = (batch_size, count)
         else:
             shape = (count,)
-        result_dict[name] = initialize_array(output_type.value, name, shape=shape, empty=True)
+        attributes = output_component_types[name]
+        if attributes is None:
+            result_dict[name] = initialize_array(output_type.value, name, shape=shape, empty=True)
+        elif attributes in [[], set()]:
```

Comment:
can't you check whether it's empty in a different way?

Reply:
@mgovers why is this not sufficient?

Reply:
Check for `isinstance` and then for contents?

Reply:
So `isinstance(attributes, (set, list) and len(attributes) == 0`. Both would do, but is there a reason any of one should be preferred?

Reply:
Prevents not creating new objects

Reply:
I also believe (please verify) that since python 3.10 also types like `Mapping` can be used in `isinstance`, which ensures you don't have to do 2 checks

Reply:
Yes, but the earlier type hint was specially list or set. Should we extend it to sequence?

> I also believe (please verify) that since python 3.10 also types like `Sequence` can be used in `isinstance`, which ensures you don't have to do 2 checks

Reply:
hmm good call. i believe so but maybe only after #465